### DISCUSSION TITLE
Capturing the enter key

### DISCUSSION THREAD

**Discussion by JonathanFarina** [danielsaidi/RichTextKit#160] [Q&A]
Hi - new to this but hoping someone can help.  

I would like to use the RichTextEditor to write a chat app.  Therefore in my app, I want to capture certain key presses.  Eg: Shift-Enter to allow me to move the curser to the new line and Enter to "send" the message, effectively run a prescribed bit of code.  I've skimmed the documentation and can't seem to find anything, so asking here for help

Thanks

**Comment by danielsaidi** [comment]
Hi @JonathanFarina 

I haven't tried this, but is it possible to add native SwiftUI key press modifiers to the text editor for this purpose?

**Comment by S1D1T1** [comment]
I too would like to intercept keypresses from a richTextEditor view. I'm failing to get key Down events. I added the .onKeyPress handler as discussed here: https://github.com/danielsaidi/RichTextKit/issues/36#issuecomment-1977456159, but it is not called.

 ... unless watch for  keyUP events. Then my onKeyPress handler is called, but at that point, it's too late and the character has already been inserted into the text.

```swift
       .onKeyPress(phases: .all) { key in
         print("key pressed: \(key.debugDescription)")
         return .ignored
       }


output:
key pressed: KeyPress(.up, "a")
key pressed: KeyPress(.up, "b")
key pressed: KeyPress(.up, "c")
key pressed: KeyPress(.up, "	")
```

My intent is to catch Tab key presses to tab between fields. In my usage, there is no role for a tab character in that text. Any hooks to intercept a keydown event before RichTextEditor?

**Reply by danielsaidi** [reply]
Hi @S1D1T1 - perhaps the wrapper around the UI/AppKit views affect the key bindings? However, I noticed that some key presses behave differently across platforms, for instance return. Can you try the `.onMoveCommand` modifier instead?

**Reply by S1D1T1** [reply]
I didn't know about  .onMoveCommand(). thank you! Alas, I added it as a view modifier to my richTextEditor and it was never called.

I have an alternative for tabbing between fields - I found that Ctrl-Tab is handled by the system automatically and does tab between my RTEs
